guard/src/rules/functions/converters.rs

```diff
@@ -0,0 +1,214 @@
+                    convert_char_to_number(val, path)? as f64,
+                )))),
+                _ => {
+                    aggr.push(None);
+                }
+            },
+            _ => {
+                aggr.push(None);
+            }
+        }
+    }
+
+    Ok(aggr)
+}
+
+fn convert_char_to_number(val: &char, path: &Path) -> crate::rules::Result<u32> {
+    val.to_digit(10).ok_or(crate::Error::ParseError(format!(
+        "attempting to convert a string: {val} into a number at {path}"
+    )))
+}
+
+pub(crate) fn parse_int(args: &[QueryResult]) -> crate::rules::Result<Vec<Option<PathAwareValue>>> {
+    let mut aggr = vec![];
+    for entry in args.iter() {
+        match entry {
+            QueryResult::Literal(val) | QueryResult::Resolved(val) => match &**val {
+                PathAwareValue::String((path, val)) => {
+                    let number = match val.parse::<i64>() {
+                        Ok(i) => Some(PathAwareValue::Int((path.clone(), i))),
+                        Err(_) => {
+                            return Err(crate::Error::ParseError(format!(
+                                "attempting to convert a string: {val} into a number at {path}"
+                            )))
+                        }
+                    };
+
+                    aggr.push(number)
+                }
+                PathAwareValue::Int((path, val)) => {
+                    aggr.push(Some(PathAwareValue::Int((path.clone(), *val))))
+                }
+                PathAwareValue::Char((path, val)) => aggr.push(Some(PathAwareValue::Int((
+                    path.clone(),
+                    convert_char_to_number(val, path)? as i64,
+                )))),
+                PathAwareValue::Float((path, val)) => {
+                    aggr.push(Some(PathAwareValue::Int((path.clone(), *val as i64))))
+                }
+                _ => {
+                    aggr.push(None);
+                }
+            },
+            _ => {
+                aggr.push(None);
+            }
+        }
+    }
+
+    Ok(aggr)
+}
+
+pub(crate) fn parse_bool(
+    args: &[QueryResult],
+) -> crate::rules::Result<Vec<Option<PathAwareValue>>> {
+    let mut aggr = vec![];
+    for entry in args.iter() {
+        match entry {
+            QueryResult::Literal(val) | QueryResult::Resolved(val) => match &**val {
+                PathAwareValue::Bool((path, val)) => {
+                    aggr.push(Some(PathAwareValue::Bool((path.clone(), *val))))
+                }
+                PathAwareValue::String((path, val)) => match val.to_lowercase().as_str() {
+                    "true" => aggr.push(Some(PathAwareValue::Bool((path.clone(), true)))),
+                    "false" => aggr.push(Some(PathAwareValue::Bool((path.clone(), false)))),
+                    _ => {
+                        return Err(crate::Error::ParseError(format!(
+                            "attempting to convert a string: {val} into a boolean at {path}"
```

Comment:
Input will always be a string and we agreed not to do this for numbers like bash right?

Reply:
Yup